File: CS-Examples/02_Data/ApplySubscriptAndSuperscript.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using Spire.Xls;
using Spire.Xls.Core.Spreadsheet;

namespace ApplySubscriptAndSuperscript
{
	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, System.EventArgs e)
		{
            // Create a new workbook
            Workbook workbook = new Workbook();

            // Get the first worksheet from the workbook
            Worksheet sheet = workbook.Worksheets[0];

            // Set the text for cell B2
            sheet.Range["B2"].Text = "This is an example of Subscript:";
            // Set the text for cell D2
            sheet.Range["D2"].Text = "This is an example of Superscript:"; 

            // Set the RTF value of cell "B3" to "R100-0.06"
            CellRange range = sheet.Range["B3"];
            range.RichText.Text = "R100-0.06";

            // Create a font and set the IsSubscript property to true
            ExcelFont font = workbook.CreateFont();
            font.IsSubscript = true;
            font.Color = Color.Green;

            // Set the font for the specified range of text in cell "B3"
            range.RichText.SetFont(4, 8, font);

            // Set the RichText value of cell "D3" to "a2 + b2 = c2"
            range = sheet.Range["D3"];
            range.RichText.Text = "a2 + b2 = c2";

            // Create a font and set the IsSuperscript property to true
            font = workbook.CreateFont();
            font.IsSuperscript = true;

            // Set the font for the specified range of text in cell "D3"
            range.RichText.SetFont(1, 1, font);
            range.RichText.SetFont(6, 6, font);
            range.RichText.SetFont(11, 11, font);

            // Auto-fit the columns to adjust their widths
            sheet.AllocatedRange.AutoFitColumns();

            // Specify the name for the resulting Excel file
            String result = "Result-ApplySubscriptAndSuperscript.xlsx"; 

            // Save the modified workbook to a file
            workbook.SaveToFile(result, ExcelVersion.Version2013);

            // Dispose of the workbook object
            workbook.Dispose();

            //Launch the MS Excel file.
            ExcelDocViewer(result);
		}

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
	}
}

File: CS-Examples/02_Data/ApplySubscriptAndSuperscript.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel
Imports Spire.Xls
Imports Spire.Xls.Core.Spreadsheet

Namespace ApplySubscriptAndSuperscript
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Excel workbook object.
            Dim workbook As New Workbook()

            ' Get the first worksheet in the workbook.
            Dim sheet As Worksheet = workbook.Worksheets(0)
            ' Set the text of cell B2 as "This is an example of Subscript:".
            sheet.Range("B2").Text = "This is an example of Subscript:"
            ' Set the text of cell D2 as "This is an example of Superscript:".
            sheet.Range("D2").Text = "This is an example of Superscript:"

            ' Get the range of cell B3.
            Dim range As CellRange = sheet.Range("B3")
            ' Set the rich text value of the cell as "R100-0.06".
            range.RichText.Text = "R100-0.06"

            ' Create a new font object.
            Dim font As ExcelFont = workbook.CreateFont()
            ' Set the IsSubscript property of the font to true (for subscript).
            font.IsSubscript = True
            ' Set the color of the font to green.
            font.Color = Color.Green

            ' Apply the specified font to the range of characters from index 4 to 8.
            range.RichText.SetFont(4, 8, font)

            ' Get the range of cell D3.
            range = sheet.Range("D3")
            ' Set the rich text value of the cell as "a2 + b2 = c2".
            range.RichText.Text = "a2 + b2 = c2"

            ' Create another font object.
            font = workbook.CreateFont()
            ' Set the IsSuperscript property of the font to true (for superscript).
            font.IsSuperscript = True

            ' Apply the specified font to the range of characters at index 1.
            range.RichText.SetFont(1, 1, font)
            ' Apply the specified font to the range of characters at index 6.
            range.RichText.SetFont(6, 6, font)
            ' Apply the specified font to the range of characters at index 11.
            range.RichText.SetFont(11, 11, font)
            ' Auto-fit the column widths to fit the content.
            sheet.AllocatedRange.AutoFitColumns()
            ' Specify the output file name.
            Dim result As String = "Result-ApplySubscriptAndSuperscript.xlsx"

            ' Save the modified workbook to the specified file with Excel 2013 format.
            workbook.SaveToFile(result, ExcelVersion.Version2013)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the MS Excel file.
            ExcelDocViewer(result)
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
